src/bignum.nr

```diff
@@ -212,18 +212,17 @@ where
     unconstrained fn __batch_invert<let M: u32>(x: [Self; M]) -> [Self; M] {
         let params = Params::get_params();
         assert(params.has_multiplicative_inverse);
-        __batch_invert::<_, MOD_BITS, _>(params, x.map(|bn| Self::get_limbs_slice(bn).as_array()))
-            .map(|limbs| Self { limbs })
+        __batch_invert::<_, MOD_BITS, _>(params, x.map(|bn: Self| bn.limbs)).map(|limbs| {
```

Comment:
I hadn't appreciated you could put type annotations inside the closure `|bn: T|`. Pretty cool.